pmlc/transforms/hoisting.cc

```diff
@@ -106,11 +107,14 @@ void HoistingState::doHoisting() {
       continue;
     if (!innerEffect.hasNoEffect())
       continue;
+    // Skip constant ops
+    if (isa<pmlc::dialect::tile::ConstantOp>(innerOp))
+      continue;
     // Check if all operands (if any) are in init
     IVLOG(3, "Checking operands " << debugString(innerOp));
     bool allOperandsInInit = true;
     for (auto operand : innerOp.getOperands()) {
-      if (!maybeCrossFunc.count(operand)) {
+      if (!inInit.count(operand)) {
```

Comment:
In addition to skipping constants, we need to allow constant operands even if they are not inInit here.